presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -563,7 +598,18 @@ private void initializeAggregationBuilderIfNeeded()
             return;
         }
 
-        if (step.isOutputPartial() || !spillEnabled) {
+        boolean partialAggregationDisabled = partialAggregationController
+                .map(PartialAggregationController::isPartialAggregationDisabled)
+                .orElse(false);
+
+        if (step.isOutputPartial() && partialAggregationDisabled) {
+            aggregationBuilder = new SkipAggregationBuilder(
```

Comment:
Do we need any special handling in this clause wrt to spillEnabled ?

Reply:
+1. looking at the code with spillEnabled we are bypassing it and going with Skip . Would that cause local OOMs?